File: sdk/python/hitechcloud/resources/url_shortener.py
"""URL Shortener resource (6 endpoints)"""

from .base import BaseResource

class UrlShortenerResource(BaseResource):
    """URL Shortener: shorten URLs, manage links, stats"""

    def shorten(self, url: str, **kwargs) -> dict:
        """POST /api/url-shortener/shorten - Shorten a URL"""
        data = {"url": url, **kwargs}
        return self._http.post("/api/url-shortener/shorten", data)

    def list_links(self) -> dict:
        """GET /api/url-shortener/links - List links"""
        return self._http.get("/api/url-shortener/links")

    def get_config(self) -> dict:
        """GET /api/url-shortener/config - Get shortener config"""
        return self._http.get("/api/url-shortener/config")

    def get_stats(self) -> dict:
        """GET /api/url-shortener/stats - Get link stats"""
        return self._http.get("/api/url-shortener/stats")

    def get_link(self, link_id: int) -> dict:
        """GET /api/url-shortener/links/{id} - Get link details"""
        return self._http.get(f"/api/url-shortener/links/{link_id}")

    def delete_link(self, link_id: int) -> dict:
        """DELETE /api/url-shortener/links/{id} - Delete link"""
        return self._http.delete(f"/api/url-shortener/links/{link_id}")

File: sdk/ruby/lib/hitechcloud/resources/url_shortener.rb
module HiTechCloud
  module Resources
    class UrlShortener < Base

        # Shorten URL
        def shorten(data)
          @http.post("/api/url-shortener/shorten", data)
        end

        # List links
        def links
          @http.get("/api/url-shortener/links")
        end

        # Get config
        def config
          @http.get("/api/url-shortener/config")
        end

        # Get stats
        def stats
          @http.get("/api/url-shortener/stats")
        end

        # Get link
        def link(id)
          @http.get("/api/url-shortener/links/%{id}" % { id: id })
        end

        # Delete link
        def delete_link(id)
          @http.delete("/api/url-shortener/links/%{id}" % { id: id })
      end
    end
  end
end
